<launch>
    <arg name="bagPath" default="empty/dir"/>
    <arg name="scanMode" default="0"/>
    <arg name="bagTopic" default="unfiltered_radar_packet"/>

    <node pkg="rosbag" type="play" name="rosbag"
        args="-l $(arg bagPath) --topics /radar_left/unfiltered_radar_packet /radar_right/unfiltered_radar_packet"/>

    <group ns="radar_left">
        <node name="radar_processor" pkg="radar_driver" type="radar_processor"
            args="-s $(arg scanMode) -t /radar_left/$(arg bagTopic) "/>
        <node name="radar_visualizer" pkg="radar_driver" type="radar_visualizer"
            args="-t /radar_left/filtered_radar_packet -n radar_left" />
    </group>

    <group ns="radar_right">
        <node name="radar_processor" pkg="radar_driver" type="radar_processor"
            args="-s $(arg scanMode) -t /radar_right/$(arg bagTopic) "/>
        <node name="radar_visualizer" pkg="radar_driver" type="radar_visualizer"
            args="-t /radar_right/filtered_radar_packet -n radar_right" />
    </group>

    <node name="radar_clustering" pkg="radar_driver" type="radar_clustering" />

    <!--TRANSFORM -->
    <node pkg="tf" type="static_transform_publisher" name="radar_right_tf" args="0 -0.5 0 0 0 0 base_link radar_right_frame 100" />

    <node pkg="tf" type="static_transform_publisher" name="radar_left_tf" args="0 0.5 0 0 0 0 base_link radar_left_frame 100" />

    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find radar_driver)/rviz/radarFOVline.rviz" />

</launch>